src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -2100,6 +2102,25 @@ public void listen(EntryRemovedEvent removedEntryEvent) {
         }
     }
 
+    /**
+     * Updates the timestamp of changed entries if the feature is enabled
+     */
+    public static class UpdateTimestampListener {
```

Comment:
I would put it in a new file (we are trying to reduce the size of BasePanel and MainFrame). I know, this was one of your questions before you posted the PR; sorry for the delay and the additional work it now means.

Reply:
When thinking about whether or not to move the listener into a separate file, we decided against it as we could not find a good place to put it. Could you suggest a package/folder?

Reply:
Although this isn't perfect, the best place for the listener is in parallel to the `BasePanel` class (being the only class that uses it). So please extract the listener into a top-level class and put it in `org.jabref.gui`. When you address my other comment regarding the test, you can also reduce the visibility of the listener class from public to default.